ar71xx: update Trendnet TEW-823DRU MAC address fix

ppancha opened the pull request:
Remove not working Trendnet TEW-823DRU mac address fix from preinit/82_patch_ath10k script. Add fix to etc/hotplug.d/ieee80211/10_fix_wifi_mac by reading LAN ethernet mac address from mtd mac partition, incrementing by 1 and writing it to ath10k interface mac address.

Signed-off-by: Pramod Pancha <pancha@vill.com>

dedeckeh commented:
The patch has styles issues as the git commit subject needs to be restricted to 50 characters; the git commit  
 description is missing and needs to explain why the commit is made.
See also https://openwrt.org/submitting-patches

ppancha commented:
Does commit efb97b9 fix the issue?

ynezz commented:
@ppancha No, you need to read again and more carefully https://openwrt.org/submitting-patches as you need to fix commits in this PR, now the state is:

```
Remove not working Trendnet TEW-823DRU mac address fix from preinit/8…
ar71xx: update non-working tew-823dru mac address fix 
Merge branch 'tew823dru-mac-fix' of https://github.com/ppancha/openwrt …
```

and it should be probably just one commit:

```
ar71xx: update non-working tew-823dru mac address fix 
```

You need to work on the commit subject and commit message, I miss the answer to `Why?` do you need this changes in your commit message.

ynezz commented:
@ppancha Are you going to incorporate the comments and move forward with this PR?

ppancha commented:
I have update the commit subject and message to make it more clear as to why the fix is needed.

ynezz commented on target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -19,6 +19,9 @@ case "$board" in
 		echo $(macaddr_add $(mtd_get_mac_binary mac 8)  $(($PHYNBR - 1)) ) > /sys${DEVPATH}/macaddress
 		;;
+	tew-823dru)
+	        echo $(macaddr_add $(strings $(find_mtd_part mac) | head -n1 ) $(($PHYNBR + 1)) ) > /sys${DEVPATH}/macaddress
```
@ppancha Sorry to bother you again, but I've just re-read the patch again and it seems to me, that it would be better to use `mtd_get_mac_ascii` or `mtd_get_mac_text` from `lib/functions/system.sh` instead of this strings/head fiddling. Would it be possible?

ppancha replied:
Thanks for the feedback. Updated patch to use mtd_get_mac_text.

ppancha committed:
ar71xx: update non-working tew-823dru mac address fix

    Remove Trendnet TEW-823DRU mac address fix from preinit/82_patch_ath10k that reads MAC address from
    incorrect location. Add mac address fix to etc/hotplug.d/ieee80211/10_fix_wifi_mac (read WAN
    ethernet mac address from mac mtd partition and store in ath10k interface mac address field)

Signed-off-by: Pramod Pancha <pancha@vill.com>

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

ynezz commented:
@ppancha While doing final review locally, I've found out about following. Originally you had following:

```
echo $(macaddr_add $(strings $(find_mtd_part mac) | head -n1 ) $(($PHYNBR + 1)) ) > /sys${DEVPATH}/macaddress
```

I've asked you to use `mtd_get_mac_ascii` or `mtd_get_mac_text` instead, but in your final version you've following:

```
echo $(mtd_get_mac_text mac 24) > /sys${DEVPATH}/macaddress
```

So I've merged [those two](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=blobdiff;f=target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac;h=35480962b2fcef95a1cd3e0e3bceb231eb0f64a0;hp=b58cf9b591f47c394c4c4b4a21e114e16af2da21;hb=a25d13a2df5470352d015ba3b86884b6a26ea425;hpb=38f3433420b5f1084e7739c35afc40b1340e90d6):

```
echo $(macaddr_add $(mtd_get_mac_text mac 24)  $(($PHYNBR + 1)) ) > /sys${DEVPATH}/macaddress
```

Can you please check it again?

ppancha commented:
The new offset of 24 in the mac partition is the WAN MAC address of the 823DRU. I now use this value as the ath10k MAC address. Previously I incremented the LAN MAC address by 1 to obtain the WAN MAC address. I don't believe this will break anything but is not required.

ynezz commented:
@ppancha We should probably use what manufacturer used, so based on this previous fragment `mtd_get_mac_ascii nvram wlan1_mac`, where they've probably used unique MAC address for WLAN I'm more inclined towards that WAN MAC + 1 solution.

@obsy You seem to have submitted support for this device, could you please provide some input on this? Thanks.

obsy commented:
I don't have this device anymore, I can not test it.

ppancha commented:
> @ppancha We should probably use what manufacturer used, so based on this previous fragment `mtd_get_mac_ascii nvram wlan1_mac`, where they've probably used unique MAC address for WLAN I'm more inclined towards that WAN MAC + 1 solution.
> 
> @obsy You seem to have submitted support for this device, could you please provide some input on this? Thanks.

The current firmware does not have anything in the nvram partition. I suspect this is why a fix was added to 82_patch_ath10k but that patch does not set a unique mac address for the 5G wlan.

ynezz commented:
> The current firmware does not have anything in the nvram partition

nvram is not part of the firmware, the content of the nvram should be initialized in the factory, so if there is nothing stored in nvram part of flash in your device, it probably means, that it wasn't initialized properly at factory or nvram content was cleared (un)intentionally.

So if  I understand this situation properly, it doesn't look like a proper fix to me anymore, it seems like a fix just for your device, which doesn't have expected content of nvram. So you've probably two options:

* fix the nvram content, so the current version of `82_patch_ath10k` in the OpenWrt tree works without modification
* apply the fix suggested in this PR locally

BTW as this is ar71xx(soon to be obsolete and you we should focus on ath79 which is ar71xx successor) it's very unlikely, that I (or anybody else) is going to look at this ever again, so I'm closing this PR now.
